ts/src/base/Exchange.ts

```diff
@@ -2506,6 +2506,17 @@ export default class Exchange {
         return res === 0;
     }
 
+    safeIntegerOmitZero (obj: object, key: IndexType, defaultValue: Int = undefined): Int {
+        const timestampString = this.omitZero (this.safeString (obj, key));
+        let timestamp = undefined;
+        if (timestampString !== undefined) {
+            timestamp = this.parseToInt (timestampString);
```

Comment:
parseToInt will crash if the value is not a number, and safeX methods are not expected to crash

Reply:
parseToInt uses `numberToString()` and I dont htink it would crash?

Reply:
@ttodua did you test it? 

```
✗ p binance numberToString "test" --sandbox
Python v3.11.7
CCXT v4.3.8
binance.numberToString(test)
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 249, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 226, in main
    result = method(*args)
             ^^^^^^^^^^^^^
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/base/decimal_to_precision.py", line 172, in number_to_string
    d = decimal.Decimal(str(x))
        ^^^^^^^^^^^^^^^^^^^^^^^
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]
```

Reply:
but i'll wait its tests

Reply:
I think you're missing the point, `numberToString` should only be used when we know for sure the input is a number, updating it to avoid a crash when providing some random string just to prevent safeIntegerOmitZero from crashing seems the wrong path to take

Reply:
ok, i'll revise this again

Reply:
@carlosmiei but I don't understand in what case this would crash, can you give some specific example?

Reply:
i replaced it with parseInt, I am not on same page, please give direct hint what method should I use instead and where, bcz I am not sure fully understand the issue

Reply:
@ttodua We're clearly not on the same page, as I said a safe method should never crash regardless of the type inside the dict/list. what happens if you do safeIntegerOmitZero (obj, 'a') and obj is `{'a': 'I'm a string because the exchange returned a weird response'}? you're calling parseInt on that which will lead to a crash on the userland

Reply:
but don't we use it on cases where we expect the timestamp numeric value? if exchange distorts the value that much faulty, then all other things can be messed up across lib.
anyway ok, i've updated, lmk if the last edit is what you meant to be.

Reply:
@ttodua The point is a safeMethod should never thrown an exception but default to undefined/defaultValue if the target is available